lib/python/qmk/keymap.py

```diff
@@ -94,3 +95,39 @@ def write(keyboard, keymap, layout, layers):
         keymap_fd.write(keymap_c)
 
     return keymap_file
+
+
+def list_keymaps(keyboard_name):
+    """ List the available keymaps for a keyboard.
+
+    Args:
+        keyboard_name: the keyboards full name with vendor and revision if necessary, example: clueboard/66/rev3
+
+    Returns:
+        a set with the names of the available keymaps
+    """
+    # parse all the rules.mk files for the keyboard
+    rules_mk = qmk.makefile.get_rules_mk(keyboard_name)
+    names = set()
+
+    if rules_mk:
+        # qmk_firmware/keyboards
+        keyboards_dir = os.path.join(os.getcwd(), "keyboards")
```

Comment:
You should probably use [pathlib](https://docs.python.org/3.5/library/pathlib.html) instead here.